examples/crowd-funding/tests/campaign_lifecycle.rs

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Integration tests for the Fungible Token application.
+
+#![cfg(not(target_arch = "wasm32"))]
+
+use crowd_funding::{CrowdFundingAbi, InitializationArgument, Operation};
+use fungible::{AccountOwner, FungibleTokenAbi};
+use linera_sdk::{
+    base::{Amount, Timestamp},
+    test::TestValidator,
+};
+use std::iter;
+
+/// Test creating a campaign and collecting pledges.
+///
+/// Creates a campaign on a `campaign_chain` and sets up the fungible token to use with three
+/// backer chains. Pledges part of each backer's balance to the campaign and then completes it,
+/// collecting the pledges. The final balance of each backer and the campaign owner is checked.
+#[tokio::test]
+async fn collect_pledges() {
+    let initial_amount = Amount::from(100);
+    let target_amount = Amount::from(220);
+    let pledge_amount = Amount::from(75);
+
+    let (validator, bytecode_id) = TestValidator::with_current_bytecode().await;
+
+    let fungible_publisher_chain = validator.new_chain().await;
+    let mut campaign_chain = validator.new_chain().await;
+    let campaign_account = AccountOwner::from(campaign_chain.public_key());
+
+    let fungible_bytecode_id = fungible_publisher_chain
+        .publish_bytecodes_in("../fungible")
+        .await;
+
+    let (token_id, backers) = FungibleTokenAbi::create_with_accounts(
+        &validator,
+        fungible_bytecode_id,
+        iter::repeat(initial_amount).take(3),
+    )
+    .await;
+
+    let campaign_state = InitializationArgument {
+        owner: campaign_account,
+        deadline: Timestamp::from(u64::MAX),
+        target: target_amount,
+    };
+    let campaign_id = campaign_chain
+        .create_application::<CrowdFundingAbi>(
+            bytecode_id,
+            token_id,
+            campaign_state,
+            vec![token_id.forget_abi()],
+        )
+        .await;
+
+    let mut pledges_and_transfers = Vec::with_capacity(2 * backers.len());
+
+    for (backer_chain, backer_account, _balance) in &backers {
+        backer_chain.register_application(campaign_id).await;
```

Comment:
This feels a little bit like cheating until we fully address #718 but I guess that's fair.

Reply:
I'm not sure I follow. You mean the way the requests for other applications are made will be different in the future? But should or could the test API stay the same?